### PR TITLE
Added hotkey for ESC key to scroll chat to bottom

### PR DESCRIPTION
**Pull request by Voiture-0**
This update introduces a hotkey function similar to Discord's. When the chat window is scrolled up, pressing `ESC` will bring the view back to the latest messages.

In DGG, if the chat is not already pinned to the bottom, pressing `ESC` will scroll it down. However, if any menus are open (e.g., emote or settings), `ESC` will first close those menus instead of scrolling. Scrolling will only occur when there are no other actions for `ESC` to perform.

### CODE REVIEW

**Review by vyneer — CHANGES_REQUESTED** [review]

File: assets/chat/js/chat.js [hunk 1]
```diff
@@ -415,7 +415,12 @@ class Chat {
 
     // ESC
     document.addEventListener('keydown', (e) => {
-      if (isKeyCode(e, KEYCODES.ESC)) ChatMenu.closeMenus(this); // ESC key
+      if (isKeyCode(e, KEYCODES.ESC)) {
+        // If any menus are open, close them first
+        if ([...this.menus].some(([, menu]) => menu.visible)) ChatMenu.closeMenus(this);
+        // If chat is not pinned, scroll to bottom
+        else if (!this.mainwindow.scrollplugin.pinned) this.mainwindow.scrollplugin.scrollBottom();
```
[the remark this answers is not in the document]
Comment (Voiture-0) [reply assets/chat/js/chat.js:422]: Oh good call! I'll use that

[the remark this answers is not in the document]
Comment (Voiture-0) [reply assets/chat/js/chat.js]: Fixed 🦃

File: assets/chat/js/chat.js [hunk 2]
```diff
@@ -415,7 +415,13 @@ class Chat {
 
     // ESC
     document.addEventListener('keydown', (e) => {
-      if (isKeyCode(e, KEYCODES.ESC)) ChatMenu.closeMenus(this); // ESC key
+      if (isKeyCode(e, KEYCODES.ESC)) {
+        const activeView = this.getActiveWindow().scrollplugin;
+        // If any menus are open, close them first
+        if (this.getActiveMenu() !== undefined) ChatMenu.closeMenus(this);
```
Comment (vyneer) [inline assets/chat/js/chat.js:421]: You can just do `this.getActiveMenu()` in the if statement without the comparison; `undefined` is falsy and the menu object is truthy, so it ends up working anyway.

**Review by vyneer — CHANGES_REQUESTED** [review]

File: assets/chat/js/chat.js (hunk 2 above)
Comment (Voiture-0) [reply assets/chat/js/chat.js]: Updated (‾◡◝)

File: assets/chat/js/chat.js [hunk 3]
```diff
@@ -415,7 +415,13 @@ class Chat {
 
     // ESC
     document.addEventListener('keydown', (e) => {
-      if (isKeyCode(e, KEYCODES.ESC)) ChatMenu.closeMenus(this); // ESC key
+      if (isKeyCode(e, KEYCODES.ESC)) {
+        const activeView = this.getActiveWindow().scrollplugin;
```
Comment (11k) [inline assets/chat/js/chat.js:419]: You can go ahead and add a method to `ChatWindow` that scrolls it to the bottom so you don't have to interface with its scroll plugin directly.

**Review by 11k — CHANGES_REQUESTED** [review]
Would you mind merging in master and making it so that if an event bar event is selected, `ESC` would unselect it? This can occur after the menu check, but before the pinned check.

Also, I'd like you to remove the comments and switch to `if` statements that aren't single-line/braceless.

File: assets/chat/js/chat.js (hunk 3 above)
Comment (Voiture-0) [reply assets/chat/js/chat.js]: Good idea! Added methods on ChatWindow to interface with its scroll plugin

**Review by 11k — APPROVED** [review]

**Pull request merged**